assets/app/js/controllers/scanDataTable.js

```diff
@@ -360,10 +359,20 @@ class ScanDataTable extends Controller {
         query.where.$and.$or = [];
         for (let i = 0; i < dataTablesInput.columns.length; i++) {
           const orSearch = {};
-          orSearch[dataTablesInput.columns[i].name] = {
-            $like: Sequelize.literal(`"%${escapedSearch}%" ESCAPE '\\'`)
-          };
-          query.where.$and.$or.push(orSearch);
+          const columnVisible = this.dataTable().columns(i);
+          if (this.getSearchPreference() === `selective-search`) {
+            if (columnVisible.visible()[0] === true) {
+              orSearch[dataTablesInput.columns[i].name] = {
+                $like: Sequelize.literal(`"%${escapedSearch}%" ESCAPE '\\'`)
+              };
+              query.where.$and.$or.push(orSearch);  
+            }
+          } else {
+            orSearch[dataTablesInput.columns[i].name] = {
+              $like: Sequelize.literal(`"%${escapedSearch}%" ESCAPE '\\'`)
+            };
+            query.where.$and.$or.push(orSearch);  
+          }
```

Comment:
Have you tested any of this?

Reply:
No currently I am not tested any of this.